File: checkers/227/main.go
package main

import (
	"fmt"
	"math"
	"testlib"
	"testlib/core"
)

func func1(a, b int, x, y float64) float64 {
	temp1, temp2 := x+float64(a), y+float64(b)
	return temp1*temp1 + temp2*temp2
}

func dx(a int, x float64) float64 {
	return 2 * (x + float64(a))
}

func dist(anX, anY, x, y float64) float64 {
	return math.Sqrt(math.Pow(anX-x, 2) + math.Pow(anY-y, 2))
}

func i2b(i_0 int) bool {
	if i_0 == 0 {
		return false
	} else {
		return true
	}
}

func createMatrix0(n int, m int) [][]int {
	matrix := make([][]int, n)
	for i := 0; i < n; i++ {
		matrix[i] = make([]int, m)
		for j := 0; j < m; j++ {
			matrix[i][j] = 0
		}
	}
	return matrix
}

func checkValueOutOfSegmen(val int, a int, b int) bool {
	if val < a || val > b {
		return true
	} else {
		return false
	}
}

func main() {
	defer func() {
		_ = testlib.GetStream(testlib.TaskOutput).Close()
		_ = testlib.GetStream(testlib.InputFromUser).Close()
	}()
	inputStream := testlib.GetReader(testlib.TaskInput)
	//outputStream := testlib.GetReader(testlib.TaskOutput)
	inputFromUserStream := testlib.GetReader(testlib.InputFromUser)
	outputToUserStream := testlib.GetWriter(testlib.OutputToUser)

	// n := core.ReadInt(inputStream)
	a := core.ReadFloat(inputStream)
	b := core.ReadFloat(inputStream)
	// ans := make([]float64, 0)
	var anX float64 = float64(-a)
	var anY float64 = float64(-b)
	var x, y float64
	cnt := 0
	var requestType byte
	for {
		requestType = core.ReadByte(inputFromUserStream)
		if cnt > 1000000000 {
			testlib.WriteAnswer(fmt.Sprintf("Too many requests"), testlib.WA)
			return
		}
		switch requestType {
		case byte('!'):
			for i := 0; i < n; i++ {
				x = core.ReadFloat(inputFromUserStream)
				y = core.ReadFloat(inputFromUserStream)
				if dist(anX, anY, x, y) > 0.1 {
					testlib.WriteAnswer(fmt.Sprintf("Incorrect user output. User answer: %.3f, %.3f. abs2: %.3f", x, y], dist(anX, anY, x, y)), testlib.WA)
					return
				} else {
					testlib.WriteAnswer(fmt.Sprintf("OK"), testlib.OK)
					return
				}
			}
		case byte('?'):
			x = core.ReadFloat(inputFromUserStream)
			y = core.ReadFloat(inputFromUserStream)
			core.WriteStringToWriter(outputToUserStream, fmt.Sprintf("%.20f %.20f %.20f\n", func1(a,b,x,y), dx(a,x), dx(b,y)))
			core.FlushWriter(outputToUserStream)
			break
		default:
			testlib.WriteAnswer(fmt.Sprintf("Wrong query: %s", string(requestType)), testlib.PE)
			return
		}
	}
}

File: checkers/227/main.cpp
#include <checkers/testlib.h>
#include <cmath>
#include <iostream>
#include <fstream>
#include <stdlib.h>
#include <algorithm>
#include <vector>
#include <iomanip>
#include <sstream>

using namespace NTestlib;

double func(int a, int b, double x, double y)
{
    return pow(x + a, 2) + pow(y + b, 2);
}
double dx(int a, double x)
{
    return 2 * (x + a);
}
double dy(int b, double y)
{
    return 2 * (b + y);
}
double dist(double anX, double anY, double x, double y)
{
    return std::sqrt(pow(anX - x, 2) + pow(anY - y, 2));
}
int main(int argc, char *argv[])
{
    InitInteractor(argc, argv);
    int a = File.ReadInt();
    int b = File.ReadInt();
    double anX = -a;
    double anY = -b;
    int cnt = 0;
    double x, y;
    while (1)
    {
        char requestType = Out.ReadChar();
        if (cnt > 1000000000)
            QuitWith(WA, "Too many requests");
        switch (requestType)
        {
        case '!':
        {
            x = Out.ReadDouble();
            y = Out.ReadDouble();
            if (dist(anX, anY, x, y) > 0.1)
            {
                std::stringstream msg;
                msg << "Incorrect user output. User answer: " << x << ", " << y << ". abs2: " << dist(anX, anY, x, y);
                QuitWith(WA, msg.str());
            }
            else
                QuitWith(AC, "OK");
            break;
        }
        case '?':
        {
            x = Out.ReadDouble();
            y = Out.ReadDouble();
            std::cout << std::setprecision(20) << func(a, b, x, y) << " " << dx(a, x) << " " << dy(b, y) << std::endl;
            break;
        }
        default:
        {
            std::stringstream msg;
            msg << "Wrong query: " << requestType;
            QuitWith(PE, msg.str());
            break;
        }
        }
        Out.Match('\n');
        cnt++;
    }
    std::cerr << "INT\n";
    return 0;
}